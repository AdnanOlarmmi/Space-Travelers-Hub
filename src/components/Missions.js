import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Mission from './Mission';
import { fetchMissions } from '../redux/missions/missionSlice';
import styles from '../App.module.css';

const Missions = () => {
  const { missions } = useSelector((state) => ({ ...state.missionReducer }));
  const dispatch = useDispatch();

  if (missions.length === 0) {
    dispatch(fetchMissions());
  }

  return (
    <table className={styles.mission_table_wrapper}>
      <thead>
        <tr className={styles.mission_table_row}>
          <th className={styles.mission_table_head}>Mission</th>
          <th className={styles.mission_table_head}>Description</th>
          <th className={styles.mission_table_head}>Status</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((item) => (
          <tr className={styles.mission_table_rows} key={item.id}>
            <Mission
              name={item.name}
              description={item.description}
              id={item.id}
              status={item.status}
            />
          </tr>
        ))}
      </tbody>
    </table>
  );
};
export default Missions;
