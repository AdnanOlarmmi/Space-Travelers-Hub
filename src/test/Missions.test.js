import React from 'react';
import render from './Rockets.test';
import Missions from '../components/Rockets';

describe('MyRocketss', () => {
  it('render My Rockets Components', () => {
    render(<Missions />);
    expect(Missions).toMatchSnapshot();
  });
});
