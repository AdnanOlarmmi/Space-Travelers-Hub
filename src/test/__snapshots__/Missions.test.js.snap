// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if button works 1`] = `[Function]`;

exports[`MyRockets render My Rockets Components 1`] = `[Function]`;

exports[`MyRocketss render My Rockets Components 1`] = `[Function]`;

exports[`Rockets Check if button works 1`] = `[Function]`;

exports[`Rockets render My Rockets Components 1`] = `[Function]`;
